pkg/health/checker/checker.go

```diff
@@ -41,28 +69,45 @@ type healthStore interface {
 	GetServiceHealth(service string) (map[string]consul.WatchResult, error)
 }
 
-type healthKV interface {
-	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
+type ReplicationControllerStore interface {
+	Get(id rcfields.ID) (rcfields.RC, error)
 }
 
-type consulHealthChecker struct {
-	client      consulutil.ConsulClient
-	kv          healthKV
-	consulStore healthStore
+type LabelReader interface {
+	GetMatches(klabels.Selector, labels.Type) ([]labels.Labeled, error)
 }
-type consulHealth interface {
-	Node(types.NodeName, *api.QueryOptions) ([]*api.HealthCheck, *api.QueryMeta, error)
+
+type healthChecker struct {
+	healthClient   HealthClient
+	resourceClient ResourceClient
+	consulClient   consulutil.ConsulClient
+	kv             healthKV
+	consulStore    healthStore
+	rcStore        ReplicationControllerStore
+	labelReader    LabelReader
 }
 
-func NewConsulHealthChecker(client consulutil.ConsulClient) ConsulHealthChecker {
-	return consulHealthChecker{
-		client:      client,
-		kv:          client.KV(),
-		consulStore: consul.NewConsulStore(client),
+func NewHealthChecker(hClient HealthClient, resourceClient ResourceClient, cClient consulutil.ConsulClient, rcStore ReplicationControllerStore, labelReader LabelReader) HealthChecker {
+	return healthChecker{
+		healthClient:   hClient,
+		resourceClient: resourceClient,
+		consulClient:   cClient,
+		kv:             cClient.KV(),
+		consulStore:    consul.NewConsulStore(cClient),
+		rcStore:        rcStore,
+		labelReader:    labelReader,
 	}
 }
 
-func (c consulHealthChecker) WatchPodOnNode(
+func NewShadowTrafficHealthChecker(cClient consulutil.ConsulClient) HealthChecker {
```

Comment:
This looks great, but you have the constructor names backwards :) this one should be the old untouched `NewConsulHealthChecker` and the other one (which does traffic shadowing to the health service) should be called `NewShadowTrafficHealthChecker`